### ISSUE TITLE
Replicate Python model in Julia

### ISSUE THREAD

**Issue by EmanuelCastanho** [JuliaAI/DecisionTree.jl#203]
I am trying to replicate a Python random forest model on Julia to do some benchmark tests.

I know it is almost impossible to have the same random forest model, because of the random nature of the build. However, I am trying to create similar models.

I am using `ensemble.RandomForestClassifier` from Python and `DecisionTree.build_forest` from Julia, but I am struggling to find the match for some hyper-parameters.

```
n_subfeatures=-1; # What is this in Python?
n_trees=140; # n_estimators in Python.
partial_sampling=0.7; # What is this in Python?
max_depth=10; # Same as in Python.
min_samples_leaf=2; # Same as in Python.
min_samples_split=2; # Same as in Python.
min_purity_increase=0; # What is this in Python? Same as min_purity_decrease?
seed=42; # random_state in Python.
class_weight="balanced_subsample"; # What is this in Julia?
bootstrap="True"; # What is this in Julia?
```

**Comment by rikhuijzer** [comment]
Hi, Emanuel. I don't know all parameters, but some I do know:

- `n_subfeatures` is the fraction of features that the trees may choose from. For example, if you set `n_subfeatures` to 1.0, then all trees may pick the same splitpoint at the root of the tree. In turn, this would make the trees more correlated and hence reduce accuracy. For details on the right choice here, see https://datascience.stackexchange.com/a/23677; although you can leave it to the default value which should be fine in most cases.
- `partial_sampling` is the fraction of samples chosen for each different tree in the random forest. For example, if you set `partial_sampling` to 1.0, then all trees will see the same data. This would make the trees way more correlated and hence it would reduce the accuracy of the forest. Generally, values between 0.6 and 0.8 are preferred and usually 0.7 is fine for all uses. I suspect that the Python implementation has a similar value or automatically tunes the value based on the sample size.

So, simply put, these two basically take a subset of the data points and features of the dataset so that every tree sees only a subset of the full dataset. In turn, seeing only a subset makes it more likely that the trees are different which increases performance. This simplification is not fully true, since the subset of the features is chosen anew for every split in each tree to ensure that the most important variable can be chosen although it may not always be at the root of the tree.

`bootstrap` is probably not a setting that you want to disable. It looks like `bootstrap: False` (Python) is the same as `partial_sampling = 1.0` (Julia).

**Comment by ablaom** [comment]
@EmanuelCastanho You happy for us to close this, as complete?

**Comment by EmanuelCastanho** [comment]
Yes, thank you
